app/services/balance_transactions/creator.rb

```diff
@@ -0,0 +1,21 @@
+class BalanceTransactions::Creator
+  Params = Struct.new(:date, :comment, :amount, :user_id)
+
+  def initialize(params)
+    @params = Params.new(params[:date], params[:comment], params[:transactions_attributes][:amount],
+      params[:user_id])
+  end
+
+  def call
+    create
+  end
+
+  private
+
+  def create
+    ActiveRecord::Base.transaction do
+      transaction = BalanceTransaction.create(date: @params[:date].to_date, comment: @params[:comment])
+      transaction.transactions.create(amount: @params[:amount], user_id: @params[:user_id])
+    end
```

Comment:
@marderer `ActiveRecord::Base.transaction` is a good thing to go with, what do you think about:
```ruby
def call
  ActiveRecord::Base.transaction do
    create_balance_transaction
    add_transaction_to_user
  end
end

private

def create_balance_transaction
  @transaction = BalanceTransaction.create(date: date.to_date, comment: comment)
end

def add_transaction_to_user
  @transaction.transactions.create(amount: amount, user_id: user_id)
end
```